core/google/cloud/_testing.py

```diff
@@ -81,5 +81,14 @@ def __init__(self, items, page_token):
         self.page_token = page_token
 
     def next(self):
-        items, self._items = self._items, None
-        return items
+        if self._items is None:
+            raise StopIteration
+        else:
+            items = self._items
+            self._items = None
```

Comment:
Given the `raise`, no need for `else:`.  E.g:

```python
        if self._items is None:
            raise StopIteration
        items, self._items = self._items, None
        return items
```

Which also leaves the swap atomic.